x-pack/plugins/apm/server/lib/helpers/metrics.ts

```diff
@@ -6,13 +6,20 @@
 
 import { getBucketSize } from './get_bucket_size';
 
-export function getMetricsDateHistogramParams(start: number, end: number) {
+export function getMetricsDateHistogramParams(
+  start: number,
+  end: number,
+  metricsInterval: number
+) {
   const { bucketSize } = getBucketSize(start, end, 'auto');
   return {
     field: '@timestamp',
 
-    // ensure minimum bucket size of 30s since this is the default resolution for metric data
-    fixed_interval: `${Math.max(bucketSize, 30)}s`,
+    // ensure minimum bucket size of configured interval since this is the default resolution for metric data
+    fixed_interval: `${Math.max(
+      bucketSize,
+      isNaN(metricsInterval) ? 30 : metricsInterval
```

Comment:
Can `metricsInterval` be `NaN`? I thought we did validation of the config and `schema.number` will ensure it's a valid number?

Reply:
Technically it can. The check is far removed from wherever this function is called though. But the real reason I added this was because it broke some tests. Let me see if I can prevent that though.

Reply:
Removed the fallback and fixed the tests.